src/filters/ws.rs

```diff
@@ -242,36 +173,27 @@ fn cvt<T>(r: tungstenite::error::Result<T>, err_message: &str) -> Poll<Result<T,
         }
     }
 }
+*/
 
 impl WebSocket {
-    pub(crate) fn from_raw_socket(
-        inner: hyper::upgrade::Upgraded,
+    pub(crate) async fn from_raw_socket(
+        upgraded: hyper::upgrade::Upgraded,
         role: protocol::Role,
         config: Option<protocol::WebSocketConfig>,
     ) -> Self {
-        let ws = protocol::WebSocket::from_raw_socket(
-            AllowStd {
-                inner,
-                context: (false, null_mut()),
-            },
-            role,
-            config,
-        );
-
-        WebSocket { inner: ws }
+        WebSocketStream::from_raw_socket(upgraded, role, config)
+            .map(|inner| WebSocket { inner })
+            .await
     }
 
     fn with_context<F, R>(&mut self, ctx: Option<&mut Context<'_>>, f: F) -> R
```

Comment:
I don't think we need `with_context` anymore, it was used to pass the `Context` to `Read` and `Write`, we can just call `Pin::new(&mut self)` or assign it to a variable if needed

Reply:
Ok, I removed `with_context` and replaced it with `Pin::new(&mut self)`